imnodes.cpp

```diff
@@ -946,6 +947,84 @@ void draw_grid(EditorContext& editor, const ImVec2& canvas_size)
     }
 }
 
+void draw_pin_shape(const EditorContext& editor, const ImVec2 pin_pos, const PinData& pin, int num_segments)
```

Comment:
The argument `num_segments` is actually being used as the third argument in each draw list function, which is the color! It is also being passed the pin color style at the callsite. So `num_segments` should be of type `ImU32` and should be called `pin_color` or something similar 🙂

Reply:
There is a `num_segments` argument in ImGui's draw list functions, but I think we can just use whatever the default value is for now.

Reply:
Done